src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -173,18 +173,129 @@ case class Task(taskId: Task.Id, runSpecVersion: Timestamp, status: Task.Status)
 object Task {
   private val log = LoggerFactory.getLogger(getClass)
 
-  case class Id(idString: String) extends Ordered[Id] {
+  /**
+    * Base for all task identifiers.
+    */
+  trait Id extends Ordered[Id] {
```

Comment:
Do we really need a common trait for ids?

Reply:
Yes. I don't see another way modelling all ids that could be part of a list or map. I'm open to other suggestions.